probe-blue/RProbe.bsv

```diff
@@ -0,0 +1,29 @@
+import "BDPI" function Bit#(n) get(Bit#(32) id, Bit#(32) cycles, Bit#(32) size);
+import "BDPI" function Action put(Bit#(32) id, Bit#(32) cycles, Bit#(n) data, Bit#(32) size);
+
+
+interface RProbe#(type get_t, type put_t);
+    method get_t get_data();
+    method Action put_data(put_t data);
+endinterface
+
+module mkRProbe#(Bit#(32) id)(RProbe#(get_t, put_t)) provisos(Bits#(get_t, wid_get), Bits#(put_t, wid_put));
+    Bit#(32) get_size = fromInteger(valueOf(TDiv#(wid_get,8)));
```

Comment:
Do not use hardcoded magic numbers in your code. If you must use, add a comment to tell what does this magic number mean. For example, you can write:

typedef 8 BYTE_WIDTH;
typedef 32 WORD_WIDTH;